Lib/asyncio/proactor_events.py

```diff
@@ -489,6 +490,9 @@ def __init__(self, proactor):
         self._accept_futures = {}   # socket file descriptor => Future
         proactor.set_loop(self)
         self._make_self_pipe()
+        self_no = self._csock.fileno()
+        if isinstance(self_no, int):
```

Comment:
I'm curious when `fileno()` doesn't return `int`?
Especially taking into account that `self._csock` is created by asyncio itself, it is not a user-provided object.

Reply:
I say this happening when _csock was mocked.

Reply:
We can fix all mocks from asyncio test suite and don't care if a third-party library will use mocking incorrectly,